src/ast/passes/macro_expansion.cpp

```diff
@@ -83,6 +84,15 @@ void MacroExpander::visit(AssignVarStatement &assignment)
     // BEGIN { $a = 1; print({let $$add_one_$a = $a + 1; $$add_one_$a + 1}); }
     return;
   }
+
+  if (auto it = arg_vars_no_mutation_.find(var->ident);
+      it != arg_vars_no_mutation_.end()) {
+    it->second->addError()
+        << "Macro '" << macro_stack_.back() << "' assigns to parameter '"
+        << var->ident << "'. Meaning it expects a variable; not an expression.";
```

Comment:
Nit: `Meaning it expects ...` is a sentence fragment, this should be a part of a longer sentence, no? E.g. `.. assigns to parameter ..., meaning it expects a variable, not an expression`. I think the semi-colon is also incorrect, since it is supposed to connect two independent clauses -- `not an expression` lacks the basic requirements for a sentence ;)

Reply:
I agree with you about the semi-colon being wrong - good catch. 

I also agree that `Meaning it expects...` is a sentence fragment but that's just my poetic nature 🖊️  - I'll update both. I'm just fussy when I'm the one not doing the grammar correcting.